sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/zapwrapper/config/ZapScanContextFactoryTest.java

```diff
@@ -585,6 +585,82 @@ void script_login_file_and_template_definition_in_sechub_config_set_but_with_wro
         assertEquals(expectedErrorMessage, exception.getMessage());
     }
 
+    @Test
+    void no_pac_file_specified_results_in_no_pac_file_configured_and_env_reader_called_once() throws ZapWrapperContextCreationException {
+        /* prepare */
+        CommandLineSettings settings = createSettingsMock();
+
+        /* execute */
+        ZapScanContext zapScanContext = factoryToTest.create(settings);
+
+        /* test */
+        assertNull(zapScanContext.getPacFilePath());
+        verify(settings).getPacFilePath();
+        verify(envVariableReader).readAsString(ZAP_LOGIN_PAC_FILE_PATH);
+    }
+
+    @Test
+    void pac_file_in_cli_settings_results_in_pac_file_configured_and_env_reader_never_called() throws ZapWrapperContextCreationException {
+        /* prepare */
+        CommandLineSettings settings = createSettingsMock();
+        String pacFilePath = "src/test/resources/example-pac-files/test-proxy.pac";
+        when(settings.getPacFilePath()).thenReturn(pacFilePath);
+
+        /* execute */
+        ZapScanContext zapScanContext = factoryToTest.create(settings);
+
+        /* test */
+        assertEquals(pacFilePath, zapScanContext.getPacFilePath().toString());
+        verify(settings).getPacFilePath();
+        verify(envVariableReader, never()).readAsString(ZAP_LOGIN_PAC_FILE_PATH);
+    }
+
+    @Test
+    void pac_file_only_in_env_variable_results_in_pac_file_configured_from_env_variable() throws ZapWrapperContextCreationException {
+        /* prepare */
+        CommandLineSettings settings = createSettingsMock();
+        String pacFilePath = "src/test/resources/example-pac-files/test-proxy.pac";
+        when(envVariableReader.readAsString(ZAP_LOGIN_PAC_FILE_PATH)).thenReturn(pacFilePath);
+
+        /* execute */
+        ZapScanContext zapScanContext = factoryToTest.create(settings);
+
+        /* test */
+        assertEquals(pacFilePath, zapScanContext.getPacFilePath().toString());
+        verify(settings).getPacFilePath();
+        verify(envVariableReader).readAsString(ZAP_LOGIN_PAC_FILE_PATH);
+    }
+
+    @Test
+    void pac_file_configured_in_cli_settings_but_does_not_exist_throws_exception() throws ZapWrapperContextCreationException {
+        /* prepare */
+        CommandLineSettings settings = createSettingsMock();
+        String pacFilePath = "not-existing-file.pac";
+        when(settings.getPacFilePath()).thenReturn(pacFilePath);
+
+        /* execute + test */
+        ZapWrapperContextCreationException exception = assertThrows(ZapWrapperContextCreationException.class, () -> factoryToTest.create(settings));
+
+        assertEquals("A pac file was specified for script login, that does not exist on the filesystem!", exception.getMessage());
```

Comment:
Just a remark: if you change the exception (i suggested before). You  have to change these tests also abit. Or you test only by contains()....